### PR TITLE
NET-229: Fix Memleak(s)

### PR DESCRIPTION
**Pull request by timoxley**
Fixes for #233.

Found two definite leaks, and a few minor potential leaks.
Resend's `last` option limits to 10,000 messages, but no such limit for resend `from`/`to`.
Resend limits for `from`/`to` with some kind of pagination like this stale PR would probably help a lot: https://github.com/streamr-dev/broker/pull/190

----

CI failure due to production resends being broken.

Might have to figure out a good way to replay large datasets in test environment, it's quite possible my testing of this yesterday caused production resends to fall over. Of course this shouldn't happen at all, and we probably should do something to break up resends into smaller chunks, for the benefit of both producer and consumer.

### CODE REVIEW

File: src/utils/PushQueue.ts [hunk 1]
```diff
@@ -247,16 +236,23 @@ export default class PushQueue<T> {
     async _cleanup() {
         this.finished = true
         const { error } = this
-        for (const p of this.nextQueue) {
+        const queue = this.nextQueue
+        this.error = undefined
+        this.nextQueue = []
+        this.pending = 0
+        this.buffer.length = 0
+        while (queue.length) {
```
Comment (teogeb) [inline src/utils/PushQueue.ts:244]: Could be `this.buffer = []`, and `while (queue.length > 0)`

File: src/utils/PushQueue.ts [hunk 2]
```diff
@@ -247,16 +236,23 @@ export default class PushQueue<T> {
...
+            const p = queue.shift()
+            if (!p) { continue }
```
Comment (teogeb) [inline src/utils/PushQueue.ts:246]: Compare explicitly to `undefined`, so that the queue can contain falsy values.

**Review by teogeb — APPROVED** [review]

File: src/utils/PushQueue.ts (hunk 2 above)
Comment (timoxley) [reply src/utils/PushQueue.ts:246]: Good thought, though the "queue" here doesn't contain values, it's a queue of promises generated by calls to `next()` while the current async `next` is inflight i.e. should never contain anything but `Defer` promises. The `this.buffer` is what contains the buffered values, though given this class is called `PushQueue` I can see the confusion. I'll rename the `queue` variable to `nextQueue`

**Comment by timoxley** [comment]
CI failure seems unrelated, merging so we can put out a new, fixed 5.x release.

**Pull request merged**